xlators/storage/posix/src/posix-entry-ops.c

```diff
@@ -412,6 +412,45 @@ posix_set_gfid2path_xattr(xlator_t *this, const char *path, uuid_t pgfid,
     return ret;
 }
 
+static int
+posix_acl_xattr_set(const char *path, dict_t *xattr_req)
+{
+    int ret = 0;
+    data_t *data = NULL;
+
+    if (!xattr_req)
+        goto out;
+
+    data = dict_get_sizen(xattr_req, POSIX_ACL_ACCESS_XATTR);
+    if (data) {
+        ret = sys_lsetxattr(path, POSIX_ACL_ACCESS_XATTR, data->data, data->len,
+                            0);
+#ifdef __FreeBSD__
+        if (ret != -1) {
+            ret = 0;
+        }
+#endif /* __FreeBSD__ */
+        if (ret != 0)
+            goto out;
+    }
```

Comment:
Could you move this to a function and call it twice from here ? (next block of code is identical to this one)

Reply:
I made the change you asked, but now that I think about it, shouldn't this FreeBSD stuff be in sys_lsetxattr? Then we probably don't need an extra function. Let me know if you agree. I will move this code to FreeBSD.

Reply:
Maybe not. Not sure what effect that will have on the syscall of fuse mount. For now I will just do what you suggested.

Reply:
It seems that FreeBSD can return > 0 in case of success, while Linux always returns 0. I think it should be fine to put it `sys_lsetxattr()`, but given that we can't easily check that it's correctly interpreted everywhere, probably it would be better to leave it here.